allows you to "rescue" a program from an exception, so that
the program completes running
